mutis/signal.py

```diff
@@ -109,70 +119,102 @@ def OU_fit(self, bins=None, rang=None, a=1e-5, b=100):
             # print('pcov: ')
             # print(np.sqrt(np.diag(pcov)))
             x_c = np.linspace(1e-5, 1.1 * np.max(x), 1000)
-            plt.plot(x_c, self.pdf(x_c, *popt), "k-", label="curve_fit", alpha=0.8)
-            res["curve_fit"] = (popt, np.sqrt(np.diag(pcov)))
+            plt.plot(x_c, self.pdf(x_c, *popt), 'k-', label='curve_fit', alpha=0.8)
+            res['curve_fit'] = (popt, np.sqrt(np.diag(pcov)))
         except Exception as e:
+            popt, pcov = None, None
             raise Exception(f"Some error fitting with curve_fit {e}")
 
+        # fit pdf with MLE
         # TODO: place outside as a helper class
-        #       and understand how it is used
-        #       is it mu really needed ?
         #       mu here is different than self.mu
-        # fit pdf with MLE
         class OU(scipy_stats.rv_continuous):
             def _pdf(self, x, l, mu):
                 return (l * mu) ** (1 + l) / scipy_special.gamma(1 + l) * np.exp(-l * mu / x) / x ** (l + 2)
 
         try:
-            fit = OU(a=a, b=np.percentile(self.values, b)).fit(self.values, 1, 1, floc=0, fscale=1)
+            fit = OU(a=a, b=np.percentile(y, b)).fit(y, 1, 1, floc=0, fscale=1)
             # print('MLE fit: (l, mu)')
             # print(fit)
             x_c = np.linspace(0, 1.1 * np.max(x), 1000)
-            plt.plot(x_c, self.pdf(x_c, fit[0], fit[1]), "k-.", label="MLE", alpha=0.8)
-            res["MLE_fit"] = fit[:-2]
+            plt.plot(x_c, self.pdf(x_c, fit[0], fit[1]), 'k-.', label='MLE', alpha=0.8)
+            res['MLE_fit'] = fit[:-2]
         except Exception as e:
             raise Exception(f"Some error fitting with MLE {e}")
+
         plt.legend(loc="lower right")
+
         plt.show()
 
-        # estimate theta
-        res["th_est1"] = fit[0] * sigma_est ** 2 / 2
-        res["th_est2"] = popt[0] * sigma_est ** 2 / 2
+        # estimate theta (from curve_fit)
+        try:
+            res['th_est1'] = fit[0] * sigma_est ** 2 / 2
+        except:
+            res['th_est1'] = None
+
+        # estimate theta (from MLE)
+        try:
+            res['th_est2'] = popt[0] * sigma_est ** 2 / 2
+        except:
+            res['th_est2'] = None
 
         return res
 
-    def OU_check_gen(self, theta, mu, sigma):
-        """Description goes here."""
+    def OU_check_gen(self, theta, mu, sigma, fpsd='lombscargle', ax1=None, ax2=None, ax3=None):
```

Comment:
is it really really needed to have axes as inputs? I believe no, but if you think it is needed for a few use cases I would suggest to work with a data structure instead. (i.e. dict), so we can short the function signature. Simple function signatures are better than long ones.. 

```Python
def OU_check_gen(self, theta, mu, sigma, fpsd='lombscargle',  **axes):

        ax1 = axes.get("ax1", None)
        ax2 = axes.get("ax2", None)
        ax3 = axes.get("ax3", None)

        if not len(axes):
            fig, (ax1, ax2, ax3) = plt.subplots(nrows=1, ncols=3, figsize=(20, 4))
```

If you think it will be needed for more than a few cases, then we can go as you propose now and develop a specific plot object for `check_gen` later that will passed to this function, instead of the axes..

Reply:
I think it is good to have the axes as inputs, and to return them, in case you need to customize the plot. I don't understand what you mean by an specific plot object, could you briefly elaborate so I get an idea?

Reply:
what I mean with this is an object with specific properties for a **check_gen** plot (i.e. exactly three `ax` in a row, the second always an histogram, the third with two different Y axes, etc.. ), so if you want to customize the plot you use simple methods defined on that object instead of more complex matplotlib generic functions.

this can be done later in another PR